mne/beamformer/tests/test_rap_music.py

```diff
@@ -0,0 +1,175 @@
+# Authors: Yousra Bekhti <yousra.bekhti@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+import numpy as np
+
+import warnings
+from nose.tools import assert_true
+from numpy.testing import assert_array_almost_equal
+
+
+import mne
+from mne.datasets import testing
+from mne.beamformer import rap_music
+from mne.utils import slow_test
+
+
+data_path = testing.data_path(download=False)
+fname_raw = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc_raw.fif')
+fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis_trunc-cov.fif')
+fname_fwd = op.join(data_path, 'MEG', 'sample',
+                    'sample_audvis_trunc-meg-eeg-oct-4-fwd.fif')
+fname_event = op.join(data_path, 'MEG', 'sample',
+                      'sample_audvis_trunc_raw-eve.fif')
+
+warnings.simplefilter('always')  # enable b/c these tests throw warnings
+
+
+def read_forward_solution_meg(fname_fwd, ch_names, **kwargs):
+    fwd = mne.read_forward_solution(fname_fwd, **kwargs)
+    return mne.pick_types_forward(fwd, eeg=False, include=ch_names,
+                                  exclude=['MEG 2443'])
+
+
+def _get_data(tmin=-0.1, tmax=0.15, event_id=1):
+    """Read in data used in tests
+    """
+    events = mne.read_events(fname_event)
+    raw = mne.io.Raw(fname_raw, preload=True)
+
+    # Setup for reading the raw data
+    raw.info['bads'] = ['MEG 2443']  # 2 bads channels
+
+    # Set up pick list: MEG - bad channels
+    picks = mne.pick_types(raw.info, meg=True, eeg=False, stim=False,
+                           eog=True, ref_meg=False, exclude='bads')
+
+    # Read epochs
+    epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=True,
+                        picks=picks, baseline=(None, 0), preload=False,
+                        reject=dict(grad=4000e-13, mag=4e-12, eog=150e-6))
+
+    evoked = epochs.average()
```

Comment:
why not using the sample_audvis-ave.fif file and the precomputed cov and avoid recomputing it from raw?
should make the tests faster.